pointcloud/basic_octree.go

```diff
@@ -182,3 +214,63 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// CollidesWith checks if the given octree collides with the given geometry and returns true if it does.
+func (octree *BasicOctree) CollidesWith(geom spatialmath.Geometry) (bool, error) {
+	return octree.CollidesWithGeometry(geom, confidenceThreshold, buffer)
+}
+
+// DistanceFrom returns the distance from the given octree to the given geometry.
+func (octree *BasicOctree) DistanceFrom(geom spatialmath.Geometry) (float64, error) {
+	// TODO (RSDK-3743): currently implemented as the bare minimum but needs to be changed to correct implementation
+	collides, err := octree.CollidesWith(geom)
+	if err != nil {
+		return math.Inf(1), err
+	}
+	if collides {
+		return -1, nil
+	}
+	return 1, nil
+}
+
+// EncompassedBy returns true if the given octree is within the given geometry.
+func (octree *BasicOctree) EncompassedBy(geom spatialmath.Geometry) (bool, error) {
+	// TODO (RSDK-3743)
+	return false, errors.New("not implemented")
+}
+
+// SetLabel sets the label of this octree.
+func (octree *BasicOctree) SetLabel(label string) {
+	octree.label = label
+}
+
+// Label returns the label of this octree.
+func (octree *BasicOctree) Label() string {
+	return octree.label
+}
+
+// String returns a human readable string that represents this octree.
+func (octree *BasicOctree) String() string {
+	template := "octree of node type %s. center: %v, side length: %v, size: %v"
+	switch octree.node.nodeType {
+	case internalNode:
+		return fmt.Sprintf(template, "internalNode", octree.center, octree.sideLength, octree.size)
+	case leafNodeEmpty:
+		return fmt.Sprintf(template, "leafNodeEmpty", octree.center, octree.sideLength, octree.size)
+	case leafNodeFilled:
+		return fmt.Sprintf(template, "leafNodeFilled", octree.center, octree.sideLength, octree.size)
+	}
+	return ""
+}
+
+// ToPoints converts an octree geometry into []r3.Vector.
+func (octree *BasicOctree) ToPoints(resolution float64) []r3.Vector {
+	// TODO (RSDK-3743)
```

Comment:
[nit] Being pedantic at this point but we normally include a descriptor of the ticket here as well as leave it outside the function. See example in combinedInverseKinematics_windows.go

```
// CreateCombinedIKSolver is not supported on windows.
// TODO(RSDK-1772): support motion planning on windows
func CreateCombinedIKSolver(model referenceframe.Frame, logger golog.Logger, nCPU int) (InverseKinematics, error) {
	return nil, errors.New("motion planning is not yet supported on Windows")
}
```